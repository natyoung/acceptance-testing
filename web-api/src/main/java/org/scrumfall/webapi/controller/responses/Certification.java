package org.scrumfall.webapi.controller.responses;

import lombok.AccessLevel;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;

@EqualsAndHashCode
@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE, force = true)
public class Certification {
    private final String id;
    private final String userId;
    private final String date;
    private final String name;

    public Certification(String id, String userId, String date, String name) {
        this.id = id;
        this.userId = userId;
        this.date = date;
        this.name = name;
    }
}
