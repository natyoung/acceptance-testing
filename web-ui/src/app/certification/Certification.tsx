import {Certificate} from "@/app/lib/definitions";

interface CertificationProps {
  certificate: Certificate;
}

export default function Certification(props: CertificationProps) {

  return (
    <div className="mb-32 grid text-center lg:max-w-10xl lg:w-full lg:mb-0 lg:grid-cols-1 lg:text-center">
      <div data-testid="certificate"
           className="text-black bg-white border-4 mb-4 border-black border-dashed w-3/4 place-self-center m-20">
        <div className="text-xl">
          <p className="p-5">This is to certify that</p>
          <p className="text-2xl italic" data-testid="certified-name">{props.certificate.name}</p>
          <p className="p-5">Was admitted to the certification</p>
        </div>
        <div className="text-5xl text-amber-500 p-5">Scrumfall Master</div>
        <div className="text-9xl p-10">{props.certificate.award}</div>
        <p className="p-5" data-testid="date-certified">{props.certificate.date}</p>
        <div className="text-xs text-green-700 p-10">
          Certificate ID: <span data-testid="certificate-id">{props.certificate.id}</span>
        </div>
      </div>
      <div className="text-xs text-green-700 p-20">
            <span onClick={window.print}
                  className="cursor-pointer underline text-blue-500 font-bold text-xl">Print</span>
      </div>
    </div>
  )
}
