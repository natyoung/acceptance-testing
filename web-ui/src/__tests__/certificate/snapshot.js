import {render, waitFor} from '@testing-library/react'
import Certification from '@/app/certification/page'
jest.mock('next/image')

it('renders certification page unchanged', async () => {
  const {container} = await waitFor(() => {
    return render(<Certification/>);
  })

  expect(container).toMatchSnapshot()
})
