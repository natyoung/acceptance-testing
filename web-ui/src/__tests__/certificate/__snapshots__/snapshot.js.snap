// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders certification page unchanged 1`] = `
<div>
  <main
    class="flex min-h-screen flex-col items-center justify-between"
  >
    <div
      class="mb-32 grid text-center lg:max-w-10xl lg:w-full lg:mb-0 lg:grid-cols-1 lg:text-center"
    >
      <div
        class="heroWrapper"
      >
        <div
          class="imageWrapper"
        />
      </div>
      <form
        class="bg-dark shadow-md rounded px-8 pt-6 pb-8 mb-4"
      >
        <h1
          class="text-4xl"
        >
          Certification
        </h1>
        0
        <div
          class="mb-4"
        >
          <label
            class="block text-blue-500 text-sm font-bold mb-2"
            for="name"
          >
            Scrumfall Master's name:
          </label>
          <input
            class="shadow appearance-none border rounded py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
            data-testid="name"
            name="name"
            type="text"
          />
        </div>
        <button
          class="bg-blue-800 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
          data-testid="buy"
          type="submit"
        >
          Buy
        </button>
      </form>
    </div>
  </main>
</div>
`;
